sphinx_js/__init__.py

```diff
@@ -9,6 +10,68 @@
 from .jsdoc import Analyzer as JsAnalyzer
 from .typedoc import Analyzer as TsAnalyzer
 
+
+# Cache this to guarantee it only runs once.
+@lru_cache(maxsize=None)
+def fix_js_make_xref():
+    """Monkeypatch to fix sphinx.domains.javascript TypedField and GroupedField
+
+    Fixes https://github.com/sphinx-doc/sphinx/issues/11021
+
+    """
+    from docutils import nodes
+
+    from sphinx.domains import javascript
+    from sphinx.locale import _
+
+    class JSXrefMixin:
+        def make_xref(
+            self,
+            rolename,
+            domain,
+            target,
+            innernode=nodes.emphasis,
+            contnode=None,
+            env=None,
+            inliner=None,
```

Comment:
Because this defines inliner and location, if we want this fix to work with Sphinx < 4.1.0, we're going to have to detect the Sphinx version and then have a different mixin for < 4.1.0.

Do we need to support Sphinx < 4.1.0? Should we do this?

Reply:
I'm not aware of a reason to support older versions of Sphinx